pallets/asset/src/lib.rs

```diff
@@ -657,6 +686,80 @@ decl_module! {
             Ok(())
         }
 
+        /// Creates a checkpoint schedule. Can only be called by the token owner primary or
+        /// secondary key. Only one schedule is allowed for an asset. In order to change an existing
+        /// schedule, it must be removed first.
+        ///
+        /// # Arguments
+        /// * `origin` - A primary or secondary key of the asset owner.
+        /// * `ticker` - The asset ticker.
+        /// * `schedule` - The checkpoint schedule to be assigned to the asset.
+        ///
+        /// # Errors
+        /// * `Unauthorized` - The caller does not own the asset.
+        /// * `CheckpointScheduleAlreadyExists` - A schedule already exists.
+        //
+        // TODO: Make the weight inversely proportional to the length of the period?
```

Comment:
Since this is something that benchmarks would not be able to help us with, would it make sense to come up with a speculative weight computation just in this PR? I don't have an opinion on how the length of a period is going to affect performance or block times. At the moment my impression is that the difference between a constant weight coefficient and a proportion is rather cosmetic.

Reply:
Instead of messing with weights here, maybe we can charge a protocol fee to dissuade the attackers? The protocol fee should be strictly more than the weight/fee these schedules can consume over the next 10(?) years.

Reply:
To be honest, as long as we set the minimum period to something relatively high, this is not an attack vector.
1) There can only be one schedule per asset and It costs 10k POLYX to create an asset.
2) Creating checkpoints is not a very computationally heavy task.

Reply:
1. We have a requirement for a schedule per pair (identity, asset). Which would allow lots of schedules for a given asset.
2. True. However we have a vector of `CheckpointRecord` which grows over time and gets decoded and re-encoded every time a due checkpoint is recorded. We might add a method to delete old checkpoints or archive old checkpoints into a rarely used vector automatically in large batches.

Reply:
Fixed by adding the pallet constant `MinCheckpointDurationSecs` and a check that every schedule has periods of at least that duration.

Reply:
We decided to go with a single op fee when a schedule is created and remove per-checkpoint op fees. Which brings the original question about a single proportional op fee or weight.